### PR TITLE
Rewrite classes 3

### PR DESCRIPTION
**Pull request by drillprop**
#40

### CODE REVIEW

File: components/questions/mobileActionButtons/MobileActionButtons.tsx [hunk 1]
```diff
@@ -12,60 +12,57 @@ interface MobileActionButtonsProps {
   justDownload: boolean;
 }
 
-class MobileActionButtonsComponent extends React.Component<
+const MobileActionButtonsComponent: React.FC<
```
Comment (mmiszy) [inline components/questions/mobileActionButtons/MobileActionButtons.tsx:15]: `React.FC` is not needed if `memo` is used

File: components/questions/questionsList/QuestionsList.tsx [hunk 2]
```diff
@@ -9,51 +9,58 @@ import { AnimateHeight } from '../../animateProperty/AnimateProperty';
...
+type QuestionListProps = {
+  selectable?: boolean;
+  unselectable?: boolean;
+  editable?: boolean;
+  questions?: AppState['questions'];
+  selectedQuestionIds?: Array<Question['id']>;
+} & {
+  toggleQuestion(questionId: Question['id']): any;
+  editQuestion?(questionId: Question['id']): any;
+} & { className?: string };
```
Comment (mmiszy) [inline components/questions/questionsList/QuestionsList.tsx:21]:
```suggestion
type QuestionListProps = {
  selectable?: boolean;
  unselectable?: boolean;
  editable?: boolean;
  questions?: AppState['questions'];
  selectedQuestionIds?: Array<Question['id']>;
  toggleQuestion(questionId: Question['id']): any;
  editQuestion?(questionId: Question['id']): any;
  className?: string;
};
```

File: components/questions/questionsList/QuestionsList.tsx [hunk 3]
```diff
@@ -9,51 +9,58 @@ import { AnimateHeight } from '../../animateProperty/AnimateProperty';
...
-  typeof defaultProps & {
-    toggleQuestion(questionId: Question['id']): any;
-    editQuestion?(questionId: Question['id']): any;
-  } & { className?: string }
-> {
-  static defaultProps = defaultProps;
-
-  render() {
-    if (!this.props.questions.data) {
+const QuestionsList: React.FC<QuestionListProps> = React.memo(
```
Comment (mmiszy) [inline components/questions/questionsList/QuestionsList.tsx:23]:
```suggestion
const QuestionsList = React.memo<QuestionListProps>(
```

File: components/questions/questionsList/questionItem/QuestionItem.tsx [hunk 4]
```diff
@@ -76,9 +76,104 @@ interface QuestionContentProps {
   forwardRef?: React.RefObject<HTMLDivElement>;
 }
 
-class QuestionContent extends React.PureComponent<QuestionContentProps> {
-  render() {
-    const { question, isSelected, forwardRef } = this.props;
+export const QuestionContent: React.FC<QuestionContentProps> = memo(
```
Comment (mmiszy) [inline components/questions/questionsList/questionItem/QuestionItem.tsx:79]:
```suggestion
export const QuestionContent = memo<QuestionContentProps>(
```

File: components/questions/questionsSidebar/QuestionsSidebar.tsx [hunk 5]
```diff
@@ -1,61 +1,44 @@
...
-    );
-  }
-}
-
-const mapStateToProps = (state: AppState) => {
-  return {
-    isSidebarOpen: state.ui.isSidebarOpen,
-  };
-};
+export default () => {
```
Comment (mmiszy) [inline components/questions/questionsSidebar/QuestionsSidebar.tsx]:
```suggestion
export const QuestionsSidebar = () => {
```

File: components/questions/selectedQuestions/SelectedQuestions.tsx [hunk 6]
```diff
@@ -18,36 +16,19 @@ import QuestionsList from '../questionsList/QuestionsList';
...
-          component="div"
-        >
-          {this.renderSelectedQuestionsList()}
-        </TransitionGroup>
-      );
-    } else {
-      return <NoQuestionsSelectedInfo />;
-    }
-  }
+export default React.memo(() => {
```
Comment (mmiszy) [inline components/questions/selectedQuestions/SelectedQuestions.tsx:19]:
```suggestion
export const SelectedQuestions = React.memo(() => {
```

File: components/questions/selectedQuestions/SelectedQuestions.tsx [hunk 7]
```diff
@@ -61,32 +42,37 @@ class SelectedQuestionsComponent extends React.Component<SelectedQuestionsProps>
...
+        className={styles.selectedQuestionsContainer}
+        component="div"
+      >
+        {renderSelectedQuestionsList()}
+      </TransitionGroup>
+    );
+  } else {
+    return <NoQuestionsSelectedInfo />;
+  }
+});
```
Comment (mmiszy) [inline components/questions/selectedQuestions/SelectedQuestions.tsx:78]:
```suggestion
}, (prevProps, nextProps) => isEqual(prevProps, nextProps));
```
I don't remember why, byt it was there before
```
shouldComponentUpdate(nextProps: Readonly<SelectedQuestionsProps>): boolean {
  return !isEqual(this.props, nextProps);
}
```

**Comment by drillprop** [comment]
Fixes #40

**Pull request merged**